data/src/test/java/org/apache/iceberg/data/TestGenericRecord.java

```diff
@@ -55,7 +54,7 @@ public void testGetIncorrectClassInstance() {
     record.set(0, 10L);
 
     Assertions.assertThatThrownBy(() -> record.get(0, CharSequence.class))
-        .isInstanceOf(IllegalStateException.class)
```

Comment:
please revert the changes to this line as it was correct and we don't want to produce a diff here